src/layouts/BaseLayout/hooks/useUserAccessGate.js

```diff
@@ -31,47 +34,88 @@ function useOnboardingRedirect({ username }) {
 
 const useUserAccessGate = () => {
   const { provider } = useParams()
-  const { termsOfServicePage, defaultOrgSelectorPage } = useFlags({
-    termsOfServicePage: false,
-    defaultOrgSelectorPage: false,
-  })
-  const { data, isLoading, isSuccess } = useUser({
+  const currentRoute = useRouteMatch()
+
+  const { termsOfServicePage, defaultOrgSelectorPage, sentryLoginProvider } =
+    useFlags({
+      termsOfServicePage: false,
+      defaultOrgSelectorPage: false,
+      sentryLoginProvider: false,
+    })
+
+  const {
+    data: userData,
+    isLoading: userIsLoading,
+    isSuccess: userIsSuccess,
+    isFetching: userIsFetching,
+  } = useUser({
     suspense: false,
-    enabled: !!provider || !config.IS_SELF_HOSTED,
+    enabled: !!provider && !config.IS_SELF_HOSTED,
   })
 
-  useOnboardingRedirect({ username: data?.user.username })
+  useOnboardingRedirect({ username: userData?.user?.username })
 
-  const isGuest = !data && isSuccess
-  let showAgreeToTerms = false,
-    showDefaultOrgSelector = false
+  const {
+    data: internalUser,
+    isLoading: internalUserIsLoading,
+    isSuccess: internalUserIsSuccess,
+    isFetching: internalUserIsFetching,
+  } = useInternalUser({
+    suspense: false,
+    enabled: !config.IS_SELF_HOSTED,
+  })
+
+  const foundUser = !userData && userIsSuccess
```

Comment:
can you change this name? maybe `missingUser`

Reply:
Done